### PR TITLE
[MM-37052] Payment information test scripts for system console

### PR DESCRIPTION
**Pull request by vinod-demansol**
#### Summary

Payment information scenario's after subscription

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-37052

image

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by stevemudie — APPROVED** [review]
Looks good @vinod-demansol !

I've updated the Zephyr Scale test cases to match what you have here and linked them to the Jira ticket as well.

File: e2e/cypress/integration/enterprise/cloud/billing/payment_information_after_subscription_spec.js [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+        cy.apiRequireLicenseForFeature('Cloud');
+
+        // # Visit Subscription page
+        cy.visit('/admin_console/billing/subscription');
+
+        // * Check for Subscription header
+        cy.contains('.admin-console__header', 'Subscription').should('be.visible');
+
+        // # Click Subscribe Now button
+        cy.contains('span', 'Upgrade Now').parent().click();
```
Comment (AGMETEOR) [inline e2e/cypress/integration/enterprise/cloud/billing/payment_information_after_subscription_spec.js:29]: Thanks a lot @vinod-demansol . I am getting this failure locally (see image below).
Screenshot 2021-09-08 at 11 52 12

Comment (vinod-demansol) [reply e2e/cypress/integration/enterprise/cloud/billing/payment_information_after_subscription_spec.js]: corrected the caption of subscribe button, please rereview it

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @vinod-demansol! I ran the tests and got it all passing, though not consistent as some tests are flaky. Please see general comments. I'll take another detailed review once resolved.

**Review by AGMETEOR — APPROVED** [review]
@vinod-demansol thanks once again. I am just getting some test failures  (see image below) but it could be that I am running these tests locally
Screenshot 2021-09-13 at 14 06 21

Otherwise, apart from Saturnino's requests, looks alright to me.

File: e2e/cypress/integration/enterprise/cloud/billing/payment_information_after_subscription_spec.js [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+        cy.apiRequireLicenseForFeature('Cloud');
+
+        // # Visit Subscription page
+        cy.visit('/admin_console/billing/subscription');
+
+        // * Check for Subscription header
+        cy.contains('.admin-console__header', 'Subscription').should('be.visible');
+
+        // # Click Subscribe Now button
+        cy.contains('span', 'Subscribe Now').parent().click();
```
[the remark this answers is not in the document]
Comment (vinod-demansol) [reply e2e/cypress/integration/enterprise/cloud/billing/payment_information_after_subscription_spec.js:32]: findByRole/findByText is not suitable here so I am keeping the same existing approach here.

File: e2e/cypress/integration/enterprise/cloud/billing/payment_information_after_subscription_spec.js [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+        cy.wait('@customer');
+
+        cy.intercept('POST', '/api/v4/cloud/payment').as('payment');
+
+        cy.intercept('POST', '/api/v4/cloud/payment/confirm').as('confirm');
+
+        cy.intercept('GET', '/api/v4/cloud/customer').as('customer');
+
+        cy.intercept('GET', '/api/v4/cloud/subscription').as('subscribe');
```
[the remark this answers is not in the document]
Comment (vinod-demansol) [reply e2e/cypress/integration/enterprise/cloud/billing/payment_information_after_subscription_spec.js]: best approach is mixing the both intercept and after that wait for element and current intercept is working fine in docker environment and element is rendering with in 4 sec So please let me know if you see any inconsistency here.

File: e2e/cypress/integration/enterprise/cloud/billing/payment_information_after_subscription_spec.js [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+        cy.get('#input_state').clear().type('teststate');
+        cy.get('#input_postalCode').clear().type('4444');
+
+        // # Click Subscribe button
+        cy.get('.RHS').find('button').should('be.enabled').click();
+
+        cy.wait(['@confirm', '@subscribe']);
+
+        // * Check for success message
+        cy.contains('Great! You\'re now upgraded', {timeout: 10000}).should('be.visible');
```
[the remark this answers is not in the document]
Comment (vinod-demansol) [reply e2e/cypress/integration/enterprise/cloud/billing/payment_information_after_subscription_spec.js]: done

File: e2e/cypress/integration/enterprise/cloud/billing/payment_information_after_subscription_spec.js [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+        cy.contains('legend', 'Country').parent().find("input[type='text']").type('India{enter}');
+        cy.get('#input_address').clear().type('testaddress');
+        cy.get('#input_city').clear().type('testcity');
+        cy.get('#input_state').clear().type('teststate');
+        cy.get('#input_postalCode').clear().type('4444');
+
+        // # Click Subscribe button
+        cy.get('.RHS').find('button').should('be.enabled').click();
+
+        cy.wait(['@confirm', '@subscribe']);
```
[the remark this answers is not in the document]
Comment (vinod-demansol) [reply e2e/cypress/integration/enterprise/cloud/billing/payment_information_after_subscription_spec.js:55]: I have used both intercept and wait in the flow so please let me know if you see any flakyness in execution.

File: e2e/cypress/integration/enterprise/cloud/billing/payment_information_after_subscription_spec.js [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+        // # Click Subscribe Now button
+        cy.contains('span', 'Subscribe Now').parent().click();
+
+        cy.intercept('POST', '/api/v4/cloud/payment/confirm').as('confirm');
+
+        cy.intercept('GET', '/api/v4/cloud/subscription').as('subscribe');
+
+        // # Enter card details
+        cy.getIframeBody().find('[name="cardnumber"]').clear().type('4242424242424242');
+        cy.getIframeBody().find('[name="exp-date"]').clear().type('4242');
```
[the remark this answers is not in the document]
Comment (vinod-demansol) [reply e2e/cypress/integration/enterprise/cloud/billing/payment_information_after_subscription_spec.js]: done

File: e2e/cypress/integration/enterprise/cloud/billing/payment_information_after_subscription_spec.js [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+        cy.getIframeBody().find('[name="cvc"]').clear().type('412');
```
[the remark this answers is not in the document]
Comment (vinod-demansol) [reply e2e/cypress/integration/enterprise/cloud/billing/payment_information_after_subscription_spec.js]: done

File: e2e/cypress/support/ui/subscription_stripe.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+Cypress.Commands.add('getIframeBody', () => {
```
[the remark this answers is not in the document]
Comment (vinod-demansol) [reply e2e/cypress/support/ui/subscription_stripe.js]: done

File: e2e/cypress/support/ui/subscription_stripe.js [hunk 9]
```diff
@@ -0,0 +1,9 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (vinod-demansol) [reply e2e/cypress/support/ui/subscription_stripe.js]: done

**Review by marianunez — APPROVED** [review]
Thanks @vinod-demansol! LGTM

**Review by saturninoabril — APPROVED** [review]
Awesome, thanks @vinod-demansol!

**Pull request merged**